src/polydatum/dal.py

```diff
@@ -50,9 +266,16 @@ def _init_service(self, key, service):
         self._services[key] = service
         return service
 
-    def __getattr__(self, name):
-        assert self._data_manager.get_active_context(), 'A DataAccessContext must be started to access the DAL.'
-        return self._services[name]
+    def _call(self, path: Tuple[PathSegment, ...], *args, **kwargs):
+        return self._handler(
+            request=DalMethodRequest(
+                self._data_manager.require_active_context(), path, args, kwargs
+            )
+        )
+
+    def __getattr__(self, name: str) -> DalMethodRequester:
+        assert self._data_manager.get_active_context(), "A DataAccessContext must be started to access the DAL."
```

Comment:
Oh wait, this negates the neat feature above. Maybe we can remove this check. If we leave this check, how about use `require_active_context`?

Reply:
We were actually wondering yesterday why you didn't explicitly raise in `get_active_context`, but then defer to an AssertionError in places like this.

Reply:
`get_active_context` is used in places to see if there is an active context without causing an error.